src/FiberTraces.cc

```diff
@@ -2775,6 +2775,50 @@ namespace pfsDRPStella = pfs::drp::stella;
       }
       return -1;
     }
+    
+    template< typename ImageT, typename MaskT, typename VarianceT, typename arrayT, typename ccdImageT, int dim >
+    bool addFiberTraceToCcdArray( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
+                                  afwImage::Image< arrayT > const& fiberTraceRepresentation,
+                                  ndarray::Array< ccdImageT, 2, dim > & ccdArray ){
+
+      const PTR(const FiberTraceFunction) ftf = fiberTrace.getFiberTraceFunction();
+      const ndarray::Array< float, 1, 1 > xCenters = fiberTrace.getXCenters();
+      ndarray::Array< size_t, 2, 1 > minCenMax = pfsDRPStella::math::calcMinCenMax( xCenters,
+                                                                                    ftf->fiberTraceFunctionControl.xHigh,
+                                                                                    ftf->fiberTraceFunctionControl.xLow,
+                                                                                    ftf->fiberTraceFunctionControl.nPixCutLeft,
+                                                                                    ftf->fiberTraceFunctionControl.nPixCutRight);
+      /// Check fiberTraceRepresentation dimension
+      if ( fiberTraceRepresentation.getDimensions()[ 0 ] != ( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 ) ){
+        string message("pfs::drp::stella::math::addFiberTraceToCcdImage: ERROR: fiberTraceRepresentation.getDimensions()[ 0 ](=");
+        message += to_string( fiberTraceRepresentation.getDimensions()[ 0 ] ) + " != ( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 )(= ";
+        message += to_string( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 ) + ")";
+        cout << message << endl;
+        throw LSST_EXCEPT( pexExcept::Exception, message.c_str() );
+      }
+      if ( fiberTraceRepresentation.getDimensions()[ 1 ] != minCenMax.getShape()[ 0 ] ){
+        string message( "pfs::drp::stella::math::addFiberTraceToCcdImage: ERROR: fiberTraceRepresentation.getDimensions()[ 1 ] )(=");
+        message += to_string( fiberTraceRepresentation.getDimensions()[ 1 ] ) + ") != minCenMax.getShape()[ 0 ](=" + to_string( minCenMax.getShape()[ 0 ] ) + ")";
+        cout << message << endl;
+        throw LSST_EXCEPT( pexExcept::Exception, message.c_str() );
+      }
+      
+      int y = ftf->yCenter + ftf->yLow;
+      for (int iY = 0; iY < xCenters.getShape()[ 0 ]; ++iY){
+        for (int x = minCenMax[ iY ][ 0 ]; x <= minCenMax[ iY ][ 2 ]; ++x){
+          ccdArray[ y + iY ][ x ] += fiberTraceRepresentation.getArray()[ iY ][ x - minCenMax[ iY ][ 0 ] ];
+        }
```

Comment:
You are interpreting an image as an array for every pixel that you are setting.  This is unacceptable.

If you must use a version of this routine that accepts an ndarray::Array please rewrite this loop to use the native Image access to pixels.  It would be much better not to use ndarray here, in which case the natural interface would be Image iterators -- but even using pixel counters would be much better than this.